### DISCUSSION TITLE
请教一下，复杂的逻辑代码是写在lua还是C里？

### DISCUSSION THREAD

**Discussion by lsx1994** [cloudwu/skynet#1582] [General]
比较复杂的逻辑，比如寻路 或者 核心战斗， 你们是直接在lua里面写吗？  还是在C写注册给lua调用？
我想直接在lua里面完成这些事情，效率会不会差很多？

**Comment by aasdpiao** [comment]
战斗服如果担心效率不高.可以考虑单独起一个服做战斗服务器集群,根据负载情况增减节点,寻路一般可以考虑前端处理寻路节点.后端验证.是否需要在lua里面感觉不用先担心性能问题.做完之后如果有性能问题在考虑C服务.一般来说都是白白担心了.

**Reply by lsx1994** [reply]
我先用lua做， 游戏没火都无所谓，真要优化说明玩家已经很多了
